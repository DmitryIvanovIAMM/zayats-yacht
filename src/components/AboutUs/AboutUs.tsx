import React from 'react';
import { sectionContainerSx } from '@/components/ContactUs/ContactUs';
import { Box } from '@mui/material';

const textStyle = {
  fontWeight: 400,
  fontSize: `1.2em`,
  lineHeight: '150%'
};
export const emphasizedTextStyle = {
  fontWeight: 'bold'
};
export const centeredSectionSx = {
  ...sectionContainerSx,
  width: '90%',
  maxWidth: '90%',
  marginLeft: 'auto',
  marginRight: 'auto',

  '@media (min-width: 900px)': {
    width: '66%',
    maxWidth: '66%',
    marginLeft: 'auto',
    marginRight: 'auto'
  }
};
export const centeredSectionExtendedSx = {
  ...sectionContainerSx,
  marginLeft: 'auto',
  marginRight: 'auto'
};

export const centerItemSivStyle = {
  display: 'flex',
  justifyContent: 'center',
  marginLeft: '10px',
  marginRight: '10px'
};

const AboutUs = () => {
  return (
    <>
      <Box style={textStyle}>
        <span style={emphasizedTextStyle}>Zayats Yacht Transport</span>
        <br />
        Welcome to Zayats Yacht Transport, your trusted partner in global yacht logistics and
        transportation services. At Zayats Yacht Transport, we specialize in providing safe,
        reliable, and efficient transport solutions for yachts of all sizes. Our mission is to
        deliver unparalleled service that ensures your vessel reaches its destination securely and
        on schedule.
        <br />
        <br />
        <span style={emphasizedTextStyle}>Our Heritage</span>
        <br />
        Founded on a passion for the sea and a deep understanding of the complexities involved in
        yacht transportation, Zayats Yacht Transport has grown to become a leading name in the
        industry. Under the leadership of Vadim Yegudkin, our president who has been a prominent
        figure in the yacht transport field for over 14 years, our team combines expertise with
        personal service to manage every aspect of your yacht&#39;s journey. From the bustling ports
        of North America to the serene waters of the Mediterranean, we cover all major routes across
        the globe.
        <br />
        <br />
        <span style={emphasizedTextStyle}>Our Services</span>
        <br />
        Zayats Yacht Transport offers a comprehensive range of services tailored to meet the unique
        needs of each client. Our services include.
        <li>
          <span style={emphasizedTextStyle}>International Yacht Shipping:</span>
          &nbsp;&nbsp;We handle all logistics, from inland transportation to international shipping,
          ensuring your yacht is transported safely across oceans.
        </li>
        <li>
          <span style={emphasizedTextStyle}>Customs Clearance:</span>
          &nbsp;&nbsp;Our team takes care of all necessary customs documentation and clearance
          procedures, making international transfers as smooth as possible.
        </li>
        <li>
          <span style={emphasizedTextStyle}>Cradling and Shrink Wrapping:</span>
          &nbsp;&nbsp;To guarantee the utmost safety during transport, we provide custom cradling
          and professional shrink wrapping.
        </li>
        <li>
          <span style={emphasizedTextStyle}>Load Master Supervision:</span>
          &nbsp;&nbsp;Our experienced load masters oversee every step of the loading and unloading
          process, ensuring that each yacht is handled with care.
        </li>
        <li>
          <span style={emphasizedTextStyle}>Insurance:</span>
          &nbsp;&nbsp;We offer comprehensive marine insurance for peace of mind throughout the
          transport process.
        </li>
        <br />
        <span style={emphasizedTextStyle}>Our Commitment</span>
        <br />
        At Zayats Yacht Transport, we are committed to excellence. Our dedicated team works closely
        with each client, providing personalized service and attention to detail that ensures every
        transport is executed flawlessly. We use the latest technology and equipment to monitor each
        shipment, giving our clients peace of mind knowing their valuable assets are in safe hands.
        <br />
        <br />
        <span style={emphasizedTextStyle}>Join Us</span>
        <br />
        Whether you&#39;re relocating, attending international regattas, or exploring new waters,
        Zayats Yacht Transport is here to make your yacht transportation experience seamless and
        stress-free. Connect with us today to learn more about how we can assist with your yacht
        transport needs. Together, let&#39;s set the course for your next maritime adventure.
        <br />
        <br />
        <span style={emphasizedTextStyle}>
          Explore. Transport. Discover. With Zayats Yacht Transport.
        </span>
      </Box>
      <div style={{ paddingTop: '20px', width: '100' }}>
        <picture>
          <source media="(max-width: 600px)" srcSet={'/images/aboutus_sm.jpeg'} />
          <img
            src={'/images/aboutus.jpeg'}
            alt="about-us-image"
            width="100%"
            style={{ width: '100%', height: 'auto' }}
          />
        </picture>
      </div>
    </>
  );
};

export default AboutUs;
