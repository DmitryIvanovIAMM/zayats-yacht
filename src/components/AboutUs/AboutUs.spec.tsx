import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import AboutUs from './AboutUs';

describe('AboutUs component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<AboutUs />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders key headings and descriptive text', () => {
    render(<AboutUs />);

    expect(screen.getByText('Zayats Yacht Transport')).toBeInTheDocument();
    expect(screen.getByText('Our Heritage')).toBeInTheDocument();
    expect(screen.getByText('Our Services')).toBeInTheDocument();
    expect(screen.getByText('Our Commitment')).toBeInTheDocument();
    expect(screen.getByText('Join Us')).toBeInTheDocument();
    expect(
      screen.getByText('Explore. Transport. Discover. With Zayats Yacht Transport.')
    ).toBeInTheDocument();
  });

  it('renders services list items', () => {
    render(<AboutUs />);

    expect(screen.getByText('International Yacht Shipping:')).toBeInTheDocument();
    expect(screen.getByText('Customs Clearance:')).toBeInTheDocument();
    expect(screen.getByText('Cradling and Shrink Wrapping:')).toBeInTheDocument();
    expect(screen.getByText('Load Master Supervision:')).toBeInTheDocument();
    expect(screen.getByText('Insurance:')).toBeInTheDocument();
  });

  it('renders responsive picture with main image', () => {
    render(<AboutUs />);

    const img = screen.getByAltText('about-us-image') as HTMLImageElement;
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', '/images/aboutus.jpeg');
    expect(img).toHaveAttribute('width', '100%');

    const source = document.querySelector(
      'source[media="(max-width: 600px)"]'
    ) as HTMLSourceElement | null;
    expect(source).not.toBeNull();
    expect(source).toHaveAttribute('srcset', '/images/aboutus_sm.jpeg');
  });
});
