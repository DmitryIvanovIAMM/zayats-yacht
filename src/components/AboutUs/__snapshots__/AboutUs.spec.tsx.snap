// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AboutUs component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
    style="font-weight: 400; font-size: 1.2em; line-height: 150%;"
  >
    <span
      style="font-weight: bold;"
    >
      Zayats Yacht Transport
    </span>
    <br />
    Welcome to Zayats Yacht Transport, your trusted partner in global yacht logistics and transportation services. At Zayats Yacht Transport, we specialize in providing safe, reliable, and efficient transport solutions for yachts of all sizes. Our mission is to deliver unparalleled service that ensures your vessel reaches its destination securely and on schedule.
    <br />
    <br />
    <span
      style="font-weight: bold;"
    >
      Our Heritage
    </span>
    <br />
    Founded on a passion for the sea and a deep understanding of the complexities involved in yacht transportation, Zayats Yacht Transport has grown to become a leading name in the industry. Under the leadership of Vadim Yegudkin, our president who has been a prominent figure in the yacht transport field for over 14 years, our team combines expertise with personal service to manage every aspect of your yacht's journey. From the bustling ports of North America to the serene waters of the Mediterranean, we cover all major routes across the globe.
    <br />
    <br />
    <span
      style="font-weight: bold;"
    >
      Our Services
    </span>
    <br />
    Zayats Yacht Transport offers a comprehensive range of services tailored to meet the unique needs of each client. Our services include.
    <li>
      <span
        style="font-weight: bold;"
      >
        International Yacht Shipping:
      </span>
        We handle all logistics, from inland transportation to international shipping, ensuring your yacht is transported safely across oceans.
    </li>
    <li>
      <span
        style="font-weight: bold;"
      >
        Customs Clearance:
      </span>
        Our team takes care of all necessary customs documentation and clearance procedures, making international transfers as smooth as possible.
    </li>
    <li>
      <span
        style="font-weight: bold;"
      >
        Cradling and Shrink Wrapping:
      </span>
        To guarantee the utmost safety during transport, we provide custom cradling and professional shrink wrapping.
    </li>
    <li>
      <span
        style="font-weight: bold;"
      >
        Load Master Supervision:
      </span>
        Our experienced load masters oversee every step of the loading and unloading process, ensuring that each yacht is handled with care.
    </li>
    <li>
      <span
        style="font-weight: bold;"
      >
        Insurance:
      </span>
        We offer comprehensive marine insurance for peace of mind throughout the transport process.
    </li>
    <br />
    <span
      style="font-weight: bold;"
    >
      Our Commitment
    </span>
    <br />
    At Zayats Yacht Transport, we are committed to excellence. Our dedicated team works closely with each client, providing personalized service and attention to detail that ensures every transport is executed flawlessly. We use the latest technology and equipment to monitor each shipment, giving our clients peace of mind knowing their valuable assets are in safe hands.
    <br />
    <br />
    <span
      style="font-weight: bold;"
    >
      Join Us
    </span>
    <br />
    Whether you're relocating, attending international regattas, or exploring new waters, Zayats Yacht Transport is here to make your yacht transportation experience seamless and stress-free. Connect with us today to learn more about how we can assist with your yacht transport needs. Together, let's set the course for your next maritime adventure.
    <br />
    <br />
    <span
      style="font-weight: bold;"
    >
      Explore. Transport. Discover. With Zayats Yacht Transport.
    </span>
  </div>
  <div
    style="padding-top: 20px;"
  >
    <picture>
      <source
        media="(max-width: 600px)"
        srcset="/images/aboutus_sm.jpeg"
      />
      <img
        alt="about-us-image"
        src="/images/aboutus.jpeg"
        style="width: 100%; height: auto;"
        width="100%"
      />
    </picture>
  </div>
</DocumentFragment>
`;
